src/arr/trove/statistics.arr

```diff
@@ -244,11 +249,61 @@ fun matrix-based-multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Nu
   B_pred_fn
 end
 
-fun multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Number>) -> (List<Number> -> Number):
+fun js-based-multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Number>) -> (List<Number> -> Number):
   doc: "returns a predictor function given a list of list of independent inputs and the correspoinding list of outputs"
   MR.multiple-regression(x_s_s, y_s)
 end
 
+fun multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Number>) -> (List<Number> -> Number):
+  matrix-based-multiple-regression(x_s_s, y_s)
+end
+
+fun zip-num-lists(lol :: List<List<Number>>) -> List<List<Number>>:
+  num-cols = lol.length()
+  num-rows = lol.get(0).length()
+  for map(i from range(0, num-rows)):
+    lol.map(lam(l): l.get(i) end)
+  end
+end
+
+fun multiple-regression-coeffs(t :: Table, input-cols :: List<String>, output-col :: String):
+  doc: "returns a list of regression coefficients, given a table, a list of independent columns, and the corresponding output column"
+  # spy: t, input-cols, output-col end
+  x_s_s = zip-num-lists(input-cols.map(lam(col-name :: String): t.get-column(col-name) end))
+  # spy: x_s_s end
+  y_s = t.get-column(output-col)
+  # spy: y_s end
+  intercepts = MX.col-matrix.make(raw-array-of(1, y_s.length()))
+  # spy: intercepts end
+  mx_X = intercepts.augment(MX.lists-to-matrix(x_s_s))
+  # spy: mx_X end
+  mx_Y = MX.col-matrix.make(raw-array-from-list(y_s))
+  # spy: mx_Y end
+  MX.mtx-to-list(MX.mtx-least-squares-solve(mx_X, mx_Y))
+end
+
+fun multiple-regression-table(t :: Table, input-cols :: List<String>, output-col :: String):
```

Comment:
General comment: Didn't @schanzer implement these functions in his starter files somewhere?  (I'm pretty certain I gave him nearly this code, so it's worth double-checking...)